spec/integration/schema_parser_spec.rb

```diff
@@ -10,7 +10,15 @@
             "attributes": {
               "name": "About naming stuff and cache invalidation"
             }
-          }
+          },
+          "included": [
```

Comment:
That's why I said I'm not completely sure about how to handle this case. But consider this:
If it doesn't fail what would happen if the data of the included resource is not the same as the one already saved in our backend. The client might act on old data and the relationship in the main resource might be wrong.  Or assuming that client only includes relationship resources if it intended to create them that would mean the relationship resource is not yet present on the backend. If we already decline here we could respond faster to the client with an error.

For normal body schemas I think the argument is a bit easier to ignore unknown attributes but there where already some internal discussions before we open sourced the gem on how this gem or maybe dry-validation could raise an error if unknown attributes are provided similar to how dry-structs does with the default constructor. I had a short look into the internals of dry-validation. In theory all the components are available to construct such a validation schema but I haven't got passed prototyping that - I'm also not quite sure it would fit into what dry-validation is doing currently. Anyway we would either need to extend dry-validation with another schema type or make it fit with the purpose of the dry gem and add it upstream. With the [proposed split](https://discuss.dry-rb.org/t/plans-for-dry-validation-dry-schema-a-new-gem/215) of dry-validation this might become easier but then we might have to adapt our schema handling anyway.